docs/development.md

```diff
@@ -0,0 +1,113 @@
+# Development
+
+Developing on Group Project XBlock v2 and running tests requires, at the minimum:
+
+* python>=2.7,<3.0
+* pip>=6.0
+* node.js>=0.10
+* npm>=1.3.10 (might work with older versions, but not checked)
+
+Other dependencies are installed via `pip install -r requirements/dev.txt` and `npm install`.
+
+Group Project XBlock v2 contains a Makefile to help with most common operations: 
+
+* `make` to install everything, run tests, check code quality and collect coverage report
+* `make test_fast` to run all tests without provisioning environment (i.e. env should be already provisioned)
+* `make quality` to run all quality checks - fails fast, so does not always output all the problems. When fixing
+    quality violations make sure to run `make quality` again until clean pass.
+* `make clean` - cleans tests and coverage results.  
+* `make diff-cover` - provides coverage report only for files and lines changed in the current branch compared to 
+    master. In order to get correct results, run entire test suite with `--coverage` flag first, and make sure to update
+    `master` branch to latest version (and rebase working branch, if necessary). Only analyzes python changes so far.
```

Comment:
@e-kolpakov "Only analyzes python changes so far." Is this still true now that you added this commit: bda7c258b4d2d5e7db0efbff737f314d803e8238?